app/src/main/kotlin/com/softteco/template/data/base/ApiResult.kt

```diff
@@ -0,0 +1,65 @@
+package com.softteco.template.data.base
+
+import com.softteco.template.data.base.error.ErrorBody
+import kotlinx.coroutines.delay
+
+sealed interface ApiResult<T : Any>
+
+class ApiSuccess<T : Any>(val code: Int, val data: T) : ApiResult<T>
+class ApiError<T : Any>(
+    val code: Int,
+    val errorBody: ErrorBody?
+) : ApiResult<T>
+
+class ApiException<T : Any>(val e: Throwable) : ApiResult<T>
+
+suspend fun <T : Any> ApiResult<T>.onSuccess(
```

Comment:
It seems it's not used now. Are we going to use it?

Reply:
Yes, I think this could be useful. For example, if we don't want to return something from a repo, but just wait for one of the options. It just doesn't fit the current cases